### DISCUSSION TITLE
Logs: Do not filter default message field

### DISCUSSION THREAD

**Discussion by FliegenKLATSCH** [grafana/grafana#58405] [Data Sources]
Elasticsearch data source
(Explore-Mode)
Logs view

The field, which is configured as default message field in the data source settings, is filtered out.
It's not possible to _add_ another field in addition to the default message field (toggling the visibility of the available fields in a log message).

To be discussed: Maybe this makes sense for other data sources, to filter this out?

My use case: we have the log "message" field as default, but as soon as you want to see any meta information in _addition_ to the log line (like the environment or server name), you cannot see the default message any longer.

See https://github.com/grafana/grafana/pull/30654

**Comment by ivanahuckova** [comment]
+cc @Elfo404 
I could think of 2 solutions:

1. I am wondering, if it would make sense for Elasticsearch to "duplicate" the log message field and have it in dataFrame twice - on second position as log message, but also add it as a regular field. That way, it is going to be shown as a log message, but you would also have it in detected fields and you would be able to show it together with other fields.
2. (which I like less) is to add another button that would have _Show this field next to the message_ instead of _Show this field instead of the message_. But I think this is very similar functionality and having both is redundant.

**Comment by FliegenKLATSCH** [comment]
@ivanahuckova If there is some special logic per data source (specifically for Elasticsearch here) then the easiest approach would be to just _not_ filter the row for ES, but for all other data sources. (But maybe there are others, where it also makes sense? I don't know this.)

**Comment by aocenas** [comment]
@ivanahuckova I think that the 2 solution would actually be better. I would like to have a solution that does not need special case Elastic while at the same time showing full message in the log detail with other fields probably isn't ideal (duplication and could push the fields way down in view port). But instead of having 2 show hide button next to each field, I would rather have one button saying "Keep the message" or something like that probably here in the note we already have when filtering:
![Screenshot from 2021-03-12 11-56-14](https://user-images.githubusercontent.com/1014802/110930965-0921b080-832a-11eb-865e-44607b22eefe.png)

**Comment by gabor** [comment]
seems to be the same problem as https://github.com/grafana/grafana/discussions/42319

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
